### DISCUSSION TITLE
Live-code-synth for newbies

### DISCUSSION THREAD

**Discussion by franck-friends** [noisedeck/noisedeck-docs#33] [Questions and Issues] [answered]
Hi,

https://**pro**.noisedeck.app/ user here.

I thought that simply copying from places like Shadertoy and then pasting would work but I was too naive I guess.
Would you have simple directions to offer to a newbie like me so that I could use existing Shaders from differences places?

![image](https://user-images.githubusercontent.com/93189139/138847177-8ce618fa-d2b7-4c4c-b47f-a3b6f68e0b0b.png)
![image](https://user-images.githubusercontent.com/93189139/138847244-1b25dbab-d575-414e-a0d1-a6871e440c37.png)

**Comment by jesshewitt** [comment, marked as answer]
Making code from Shadertoy work with Noisedeck is a little complicated - you'd need to make some modifications. 

The built-in uniforms are different. You'd need to change references from **iTime** to **time**. Shadertoy uses a vec3 **iResolution**, while Noisedeck uses a vec2 **resolution**. Noisedeck doesn't have any of Shadertoy's [other uniforms](https://www.shadertoy.com/howto), so you'd need to replace those.

You'd need to rename the **mainImage** function to **main** and remove the arguments. Replace references to **C** with **gl_FragCoord.xy**. Change line 5 from **O = vec4(0);** to **vec4 O = vec4(0);** and add the line **gl_FragColor = O;** before the final curly bracket. There may be more changes needed to get the code to run.

If you're not too familiar with shaders, you may have an easier time porting examples from [The Book of Shaders](https://thebookofshaders.com/), which is also a great learning reference. You would need to change **u_resolution** to **resolution** and **u_time** to **time**, and replace **u_mouse** with another vec2.

Hope this helps!

**Reply by franck-friends** [reply]
Thanks for the explanation.
No plans for a feature allowing pasting from major online shaders collections sites?

**Reply by jesshewitt** [reply]
No, unfortunately that would be non-trivial to implement. We do plan to replace the current live code modules with user-created full modules, but it's not quite ready yet.
